---
ms.openlocfilehash: 33ad1c044001e0a8d09708cc7a1f06e05cb307de
ms.sourcegitcommit: 5b6d778ebb269ee6684fb57ad69a8c28b06235b9
ms.translationtype: HT
ms.contentlocale: de-DE
ms.lasthandoff: 04/08/2019
ms.locfileid: "59235481"
---
### <a name="different-exception-handling-for-objectcontextcreatedatabase-and-dbproviderservicescreatedatabase-methods"></a><span data-ttu-id="874d5-101">Unterschiedliche Ausnahmebehandlung für die Methoden „ObjectContext.CreateDatabase“ und „DbProviderServices.CreateDatabase“</span><span class="sxs-lookup"><span data-stu-id="874d5-101">Different exception handling for ObjectContext.CreateDatabase and DbProviderServices.CreateDatabase methods</span></span>

|   |   |
|---|---|
|<span data-ttu-id="874d5-102">Details</span><span class="sxs-lookup"><span data-stu-id="874d5-102">Details</span></span>|<span data-ttu-id="874d5-103">Ab .NET Framework 4.5 versuchen <code>CreateDatabase</code>-Methoden, eine leere Datenbank zu löschen, wenn die Datenbankerstellung fehlgeschlagen ist.</span><span class="sxs-lookup"><span data-stu-id="874d5-103">Beginning in .NET Framework 4.5, if database creation fails, <code>CreateDatabase</code> methods will attempt to drop the empty database.</span></span> <span data-ttu-id="874d5-104">Wenn dieser Vorgang erfolgreich ist, wird die ursprüngliche <xref:System.Data.SqlClient.SqlException?displayProperty=name> weitergegeben (anstelle der <xref:System.InvalidOperationException?displayProperty=name>, die in .NET Framework 4.0 immer ausgelöst wurde).</span><span class="sxs-lookup"><span data-stu-id="874d5-104">If that operation succeeds, the original <xref:System.Data.SqlClient.SqlException?displayProperty=name> will be propagated (instead of the <xref:System.InvalidOperationException?displayProperty=name> that was always thrown in .NET Framework 4.0)</span></span>|
|<span data-ttu-id="874d5-105">Vorschlag</span><span class="sxs-lookup"><span data-stu-id="874d5-105">Suggestion</span></span>|<span data-ttu-id="874d5-106">Beim Abfangen einer <xref:System.InvalidOperationException?displayProperty=name>-Ausnahme während der Ausführung von <xref:System.Data.Objects.ObjectContext.CreateDatabase> oder <xref:System.Data.Common.DbProviderServices.CreateDatabase(System.Data.Common.DbConnection,System.Nullable{System.Int32},System.Data.Metadata.Edm.StoreItemCollection)> sollten SQLExceptions-Ausnahmen nun ebenfalls abgefangen werden.</span><span class="sxs-lookup"><span data-stu-id="874d5-106">When catching an <xref:System.InvalidOperationException?displayProperty=name> while executing <xref:System.Data.Objects.ObjectContext.CreateDatabase> or <xref:System.Data.Common.DbProviderServices.CreateDatabase(System.Data.Common.DbConnection,System.Nullable{System.Int32},System.Data.Metadata.Edm.StoreItemCollection)>, SQLExceptions should now also be caught.</span></span>|
|<span data-ttu-id="874d5-107">Bereich</span><span class="sxs-lookup"><span data-stu-id="874d5-107">Scope</span></span>|<span data-ttu-id="874d5-108">Gering</span><span class="sxs-lookup"><span data-stu-id="874d5-108">Minor</span></span>|
|<span data-ttu-id="874d5-109">Version</span><span class="sxs-lookup"><span data-stu-id="874d5-109">Version</span></span>|<span data-ttu-id="874d5-110">4.5</span><span class="sxs-lookup"><span data-stu-id="874d5-110">4.5</span></span>|
|<span data-ttu-id="874d5-111">Typ</span><span class="sxs-lookup"><span data-stu-id="874d5-111">Type</span></span>|<span data-ttu-id="874d5-112">Laufzeit</span><span class="sxs-lookup"><span data-stu-id="874d5-112">Runtime</span></span>|
|<span data-ttu-id="874d5-113">Betroffene APIs</span><span class="sxs-lookup"><span data-stu-id="874d5-113">Affected APIs</span></span>|<ul><li><xref:System.Data.Objects.ObjectContext.CreateDatabase?displayProperty=nameWithType></li><li><xref:System.Data.Common.DbProviderServices.CreateDatabase(System.Data.Common.DbConnection,System.Nullable{System.Int32},System.Data.Metadata.Edm.StoreItemCollection)?displayProperty=nameWithType></li></ul>|
